CRM/Contact/Form/Search/Custom/Base.php

```diff
@@ -44,6 +44,35 @@ public function buildTaskList(CRM_Core_Form_Search $form) {
     return $form->getVar('_taskList');
   }
 
+  /**
+   * Override this when you want to implement a custom task list.
+   * By default the custom search has the same actions as a basic contact search, or
+   * an advanced search.
+   * Override this in your class and specify your own type.
+   *
+   * The type is passed the hook_civicrm_searchTasks.
+   *
+   * @return string
+   */
+  public function getObjectTypeForTaskList() {
+    return 'contact';
```

Comment:
I think we ought to stick with capitalized entity names like APIv4 uses.

Reply:
Makes sense and will implement that

Reply:
I was going to implement this but looking at `CRM_Contact_Task` I also see a lower case entity name. 
See: https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/Task.php#L42

@colemanw if we are going to use capitalized entity names we also need to solve `CRM_Contact_Task` however I do think it is behind the scope of this PR. I am happy to do a follow up PR to fix this after this one is merged. What do you think?